test/widgets/content_test.dart

```diff
@@ -252,20 +257,20 @@ void main() {
       await tester.pump();
       await tester.pump();
 
-      final headers = httpClient.request.headers.values;
-      check(authHeaders.keys).deepEquals(['Authorization']);
-      return headers['Authorization']?.single;
+      return httpClient.request.headers.values;
     }
 
     testWidgets('includes auth header if `src` on-realm', (tester) async {
-      check(await actualAuthHeader(tester, Uri.parse('https://chat.example/image.png')))
-        .isNotNull().equals(authHeaders['Authorization']!);
+      check(await actualHeaders(tester, Uri.parse('https://chat.example/image.png')))
+        ..['Authorization'].single.equals(authHeaders['Authorization']!)
+        ..['User-Agent'].single.equals(userAgentHeader()['User-Agent']!);
```

Comment:
Rather than check only these two headers, I think I'd prefer to continue to have a check that there aren't any unexpected additional headers. That can be done by a `deepEquals` on the headers map.